### PR TITLE
Update Advanced tutorials (4xx) for model deprecations and SDK API drift

### PR DESCRIPTION
**Pull request by nealwu**
Branched from #732 to split the tutorial refresh into smaller PRs.

Refreshes the Advanced tutorials (401–407). Fourth in the five-PR stack — **based on the 3xx PR (#771); review/merge that first.** Issues found by running each notebook end-to-end.

## Changes
- **401 SL Hyperparameters** — `Qwen3.5-4B`; note that the notebook shows the sweep API + a pre-computed table (it doesn't launch a live sweep, which would run full SFT jobs).
- **402 RL Hyperparameters** — `Qwen3.5-4B`.
- **403 DPO** — `Qwen3.5-4B` (`qwen3_5_disable_thinking`); note explaining the empty `<think></think>` markers in the rendered output.
- **404 Sequence Extension** — clarify that the multi-turn demo uses placeholder token IDs.
- **405 Multi-Agent** — `Qwen3.5-4B` (`qwen3_5_disable_thinking`).
- **406 Prompt Distillation** — `Qwen3.5-4B` (`qwen3_5_disable_thinking`); SDK fixes; evaluate the student on a held-out test set so the result shows generalization, not memorization; reorder the results table
so variable-width multilingual text doesn't break alignment.
- **407 Full RLHF Pipeline** — `Qwen3.5-9B-Base` base; import `TrainOnWhat` from `renderers` (it moved); `await main()` instead of `asyncio.run` in all three stages; add the now-required `recipe_name` to each `Config`; and render the final eval prompt with the model's renderer + its stop sequences (the old raw `tokenizer.encode` + `stop=["\n\n"]` made both models stop immediately and print nothing).

Ran in marimo against the live Tinker service. Diffs include marimo churn (version-string bump, trimmed cell-return tuples, imports left unused after the edits).

### CODE REVIEW

File: tutorials/406_prompt_distillation.py [hunk 1]
```diff
@@ -61,42 +61,42 @@ def _(mo):
...
-
-4) When in doubt, choose 'ot' rather than guessing.
-
-Output format:
-- Respond with EXACTLY one line: "Final Answer: xx"
-- Where xx is one of {ar, de, el, en, es, fr, hi, ru, tr, ur, vi, zh, ot} and nothing else.
-
-Text to classify:
-{text}"""
+    Goal: Classify the language of the provided text into exactly one of these labels:
```
Comment (derek-tml) [inline tutorials/406_prompt_distillation.py:67]: nit: the indentation here matters, i think the old one was more correct.

File: tutorials/403_dpo_preferences.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def _():
 @app.cell(hide_code=True)
 def _(mo):
     mo.md(r"""
-    # Tutorial 11: DPO and Preference Learning
+    # Tutorial 403: DPO and Preference Learning
 
     Build preference data, train with DPO, and evaluate with a PreferenceModel.
```
Comment (derek-tml) [inline tutorials/403_dpo_preferences.py]: looks like we removed PreferenceModel

**Review by derek-tml — APPROVED** [review]

File: tutorials/406_prompt_distillation.py (hunk 1 above)
Comment (nealwu) [reply tutorials/406_prompt_distillation.py:67]: this is tricky because marimo did this automatically. even if we fix it, users running the marimo notebook will end up with the indented version of the prompt

File: tutorials/403_dpo_preferences.py (hunk 2 above)
Comment (nealwu) [reply tutorials/403_dpo_preferences.py]: thanks, updated this description to avoid overclaiming

File: tutorials/406_prompt_distillation.py (hunk 1 above)
Comment (nealwu) [reply tutorials/406_prompt_distillation.py:67]: updating to use `textwrap.dedent()` which is robust even with marimo automatically adding indentation
